### PR TITLE
Modify TemplateTestBase to launch multiple templates

### PR DESCRIPTION
**Pull request by Deep1998**
Modified the TemplateTestBase class to support launching multiple templates while maintaining backward compatibility for single template ITs.

### CODE REVIEW

**Review by Abacn — COMMENTED** [review]
Thanks, this lgtm, just had a minor comment

File: it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java [hunk 1]
```diff
@@ -180,72 +188,115 @@ public void setUpBase() throws ExecutionException {
...
+        template = getTemplateAnnotation(templateIntegrationTest, templateClass);
+        if (template == null) {
+          return;
+        }
+        if (template.placeholderClass() != void.class) {
+          templateClass = template.placeholderClass();
+        }
+        multiTemplates.add(template);
+        multiTemplateClasses.add(templateClass);
+        pipelineLauncher = buildLauncher(templateClass, template);
```
Comment (aksharauke) [inline it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java:233]: Do we need separate pipeline launchers per template when there are multiple templates?

File: it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java [hunk 2]
```diff
@@ -283,7 +334,8 @@ private Template getTemplateAnnotation(
    * It identifies whether the template is v1 (Classic) or v2 (Flex) to setup the Maven reactor
    * accordingly.
    */
-  private String[] buildMavenStageCommand(String prefix, File pom, String bucketName) {
+  private String[] buildMavenStageCommand(
+      String prefix, File pom, String bucketName, Template template) {
```
[the remark this answers is not in the document]
Comment (Deep1998) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java]: Correct, added field comments to clarify. Also changed the method parameters.

File: it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java (hunk 1 above)
Comment (Deep1998) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java]: Added null check to do this only once. This earlier would be a redundant call re-initialising to the same value each time.

Comment (aksharauke) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java]: In the launchTemplate method, when setupShutdownHook is enabled - a separate thread is created to which the pipelineLauncher object is passed. Will there be multi-threading issues if there are multiple templates and the same pipelineLauncher object is being used across multiple threads to cancel the job?

Comment (Deep1998) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java]: I went through the beam code it looks like we should be safe. I could not find documentation explicitly saying it thread-safe though nor have i tested it.

Comment (aksharauke) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java]: Cool, thank you.

**Review by aksharauke — APPROVED** [review]

**Pull request merged**